modules/git/grep.go

```diff
@@ -71,10 +73,20 @@ func GrepSearch(ctx context.Context, repo *Repository, search string, opts GrepO
 			defer stdoutReader.Close()
 
 			isInBlock := false
-			scanner := bufio.NewScanner(stdoutReader)
+			rd := bufio.NewReaderSize(stdoutReader, util.IfZero(opts.MaxLineLength, 16*1024))
 			var res *GrepResult
-			for scanner.Scan() {
-				line := scanner.Text()
+			for {
+				lineBytes, isPrefix, err := rd.ReadLine()
+				if isPrefix {
+					lineBytes = slices.Clone(lineBytes)
+					for isPrefix && err == nil {
+						_, isPrefix, err = rd.ReadLine()
+					}
+				}
+				if len(lineBytes) == 0 && err != nil {
+					break
+				}
```

Comment:
Pity that we now have to treat `err != nil` as a successful case.
I mean, it does somewhat make sense, but at the same time it probably leads to false negative results.

Reply:
Good catch. Actually I have thought about this change and I think it should be good enough.

The real cases for returning "err" in this "grep" functions:

1. Command fails, the error is returned by "cmd.Run"
2. Pipe fails (this case), I think it would seldom happen in real life. Even if happens, returning partially parsed results is still better than showing a 500 error page.